docs/develop/go/temporal-nexus.mdx

```diff
@@ -0,0 +1,476 @@
+
+```
+temporal operator namespace create --namespace my-target-namespace
+temporal operator namespace create --namespace my-caller-namespace
+```
+
+## Create a Nexus Endpoint to route requests from caller to handler {#create-nexus-endpoint}
+
+After establishing caller and handler Namespaces, the next step is to create a Nexus Endpoint to route requests.
+
+```go
+temporal operator nexus endpoint create \
+  --name myendpoint \
+  --target-namespace my-target-namespace \
+  --target-task-queue my-handler-task-queue \
+  --description-file ./service/description.md
+```
+
+## Define the Nexus Service API contract {#define-nexus-service-api-contract}
+
+Defining a clear API contract for the Nexus Service is crucial for smooth communication between services.
+
+[View the source code](https://github.com/temporalio/samples-go/tree/main/nexus) in the context of the rest of the application code.
+
+```
+git clone https://github.com/temporalio/samples-go.git
+cd samples-go/nexus
+```
+
+The Nexus Service API contract can be in whatever form works best for your environment.
+Each [Temporal SDK includes and uses a default Data Converter](https://docs.temporal.io/dataconversion), that encodes payloads in the following order: Null, Byte array, Protobuf JSON, JSON.
+In a polyglot environment, Protobuf and JSON are common choices, but this example uses native Go types.
+
+In this example, there is a `service` package that describes the Service and Operation names along with input/output types for caller Workflows to use the Nexus Endpoint.
+
+```go
+package service
+
+const HelloServiceName = "my-hello-service"
+
+// Echo operation
+const EchoOperationName = "echo"
+
+type EchoInput struct {
+	Message string
+}
+
+type EchoOutput EchoInput
+```
+
+## Develop a Nexus Service & Operation handlers {#develop-nexus-service-operation-handlers}
+
+Nexus `Operation` handlers are typically defined in the same Worker as the underlying Temporal primitives they abstract. `Operation` handlers can decide if a given Nexus Operation will be synchronous or asynchronous, execute arbitrary code, and invoke underlying Temporal primitives such as a Workflow, Query, Signal, or Update.
+
+The `temporalnexus` package has builders to create Nexus Operations:
+
+- `NewSyncOperation` \- Simple synchronous RPC handlers, such as for Signals
+- `NewWorkflowRunOperation` \- Run a Workflow as an asynchronous Nexus Operation
+
+Start with a `NewSyncOperation` example, and then use `NewWorkflowRunOperation` to start a handler Workflow from a Nexus Operation.
+
+### Develop a Synchronous Nexus Operation handler
+
+The `temporalnexus.NewSyncOperation` builder function is for exposing simple RPC handlers. It’s handler function is provided with an SDK client that can be used for signaling, querying, and listing Workflows, but implementations are free to make arbitrary calls to other services or databases, or perform computations such as this one:
+
+```go
+package handler
```

Comment:
Looks like the formatting got broken when you pasted the sample here. Are we using snipsync?